client/components/tinymce/plugins/mentions/suggestion.jsx

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+const getRegExpFor = function( type, textToHighlight ) {
+	const expressions = {};
+
+	expressions.username = '(^' + textToHighlight + ')(\\w*)\\s*';
+	expressions.fullName = '(^.*?)(\\b' + textToHighlight + ')(.*)';
```

Comment:
If only one of these patterns is going to be used based on the incoming arguments, maybe an `if` or `switch` would be more appropriate?

```js
const getRegExpFor = function( type, textToHighlight ) {
	let expression;
	switch ( type ) {
		case 'username':
			expression = '(^' + textToHighlight + ')(\\w*)\\s*';
			break;

		case 'fullname':
			expression = '(^.*?)(\\b' + textToHighlight + ')(.*)';
			break;
	}

	return new RegExp( expression, 'ig' );
};
```